cloud/deployment/fromfile/fromfile.go

```diff
@@ -951,13 +951,16 @@ func checkEnvVars(deploymentFromFile *inspect.FormattedDeployment, action string
 	return nil
 }
 
-func createEnvVarsRequest(deploymentFromFile *inspect.FormattedDeployment) (envVars []astroplatformcore.DeploymentEnvironmentVariableRequest) {
+func createEnvVarsRequest(deploymentFromFile *inspect.FormattedDeployment, action string) (envVars []astroplatformcore.DeploymentEnvironmentVariableRequest) {
 	envVars = []astroplatformcore.DeploymentEnvironmentVariableRequest{}
 	for i := range deploymentFromFile.Deployment.EnvVars {
 		var envVar astroplatformcore.DeploymentEnvironmentVariableRequest
 		envVar.IsSecret = deploymentFromFile.Deployment.EnvVars[i].IsSecret
 		envVar.Key = deploymentFromFile.Deployment.EnvVars[i].Key
-		envVar.Value = &deploymentFromFile.Deployment.EnvVars[i].Value
+		// If the action is update, isSecret is true and value is empty, then the value should be left as nil
+		if !(action == updateAction && deploymentFromFile.Deployment.EnvVars[i].IsSecret && deploymentFromFile.Deployment.EnvVars[i].Value == "") {
```

Comment:
Does this mean, i cannot add an empty string for a secret env var? Not an ideal use-case but thinking what if i wanted to?

Reply:
yeah that would be a strange case
i don't see why you would want to have a secret variable as `""` 
anyways if that were the case, i made some updates to the code to allow nil values for no-ops and to allow `""` to be updated